lib/data/repository/dio_marvell_repository.dart

```diff
@@ -0,0 +1,107 @@
+import 'dart:convert';
+import 'dart:ui';
+import 'package:dio/dio.dart';
+import 'package:flutter/foundation.dart';
+import 'package:marvelapp_flutter/data/model/response_models/character.dart';
+import 'package:marvelapp_flutter/data/model/response_models/api_response.dart';
+import 'package:marvelapp_flutter/data/repository/marvell_repository.dart';
+import '../model/response_models/series.dart';
+
+class DioMarvellRepository extends MarvellRepository {
+  final Dio dio = Dio(
+    BaseOptions(
+        contentType: 'application/json', responseType: ResponseType.plain),
+  );
+  final String fakeTimeStamp = "12345";
+  final String publicKey = "b2bd25766ee84a0881b157960b3d3590";
+  final String hash = "f43ba4d3c12135105017b1f45993942e";
+
+  @override
+  Future<List<Character>?> getCharacters() async {
```

Comment:
In this particular case it look like `data?.results` should alwayr return some heroes, but if not we can just show the error (something like "No data error"). So the suggestion is to return non-nullable `Future<List<Character>>` in this method and in case in line 29 `data == null` return Future.error("No data error");

Reply:
> I have run your app and it looks really good!
> 
> Some suggestions I left in the comments. In addition to that we can a bit improve folder structure by putting folders `navigation`, `screens`, `widgets` into `presentation` folder.

done

Reply:
Fix case about data == null in all methods in repository. In general casese repository can't return null value at all? 
в смысле при проектировании метод репозитория может вернуть null или нет. Или должен быть механизм который если данные равны null вернет ошибку как в моем случае?